avocado/core/resolver.py

```diff
@@ -0,0 +1,174 @@
+    RETURN = object()
+    #: Continue to resolve the given reference
+    CONTINUE = object()
+
+
+class ReferenceResolution:
+
+    """
+    Represents one complete reference resolution
+
+    Note that the reference itself may result in many resolutions, or
+    none.
+    """
+
+    def __init__(self, reference, result, resolutions=None, info=None, origin=None):
+        """
+        :param reference: a specification that can eventually be resolved
+                          into a test (in the form of a
+                          :class:`avocado.core.nrunner.Runnable`)
+        :type reference: str
+        :param result: if the complete resolution was a success,
+                       failure or error
+        :type result: :class:`ReferenceResolutionResult`
+        :param resolutions: the runnable definitions resulting from the
+        resolution
+        :type resolutions: list
+        :param info: free form information the resolver may add
+        :type info: str
+        :param origin: the name of the resolver that performed the resolution
+        :type origin: str
+        """
+        self.reference = reference
+        self.result = result
+        if resolutions is None:
+            resolutions = []
+        self.resolutions = resolutions
+        self.info = info
+        self.origin = origin
+
+
+class Resolver(EnabledExtensionManager):
+
+    """
+    Main test reference resolution utility.
+
+    This performs the actual resolution according to the active
+    resolver plugins and a resolution policy.
+    """
+
+    DEFAULT_POLICY = {
+        ReferenceResolutionResult.SUCCESS: ReferenceResolutionAction.RETURN,
+        ReferenceResolutionResult.NOTFOUND: ReferenceResolutionAction.CONTINUE,
+        ReferenceResolutionResult.ERROR: ReferenceResolutionAction.CONTINUE
+    }
+
+    def __init__(self):
+        super(Resolver, self).__init__('avocado.plugins.resolver')
+
+    def resolve(self, reference):
+        resolution = []
+        for ext in self.extensions:
+            try:
+                result = ext.obj.resolve(reference)
+                if not result.origin:
+                    result.origin = ext.name
+            except Exception as exc:
+                result = ReferenceResolution(reference,
+                                             ReferenceResolutionResult.ERROR,
+                                             info=exc,
+                                             origin=ext.name)
+            resolution.append(result)
+            action = self.DEFAULT_POLICY.get(result.result,
+                                             ReferenceResolutionAction.CONTINUE)
+            if action == ReferenceResolutionAction.RETURN:
+                break
+        return resolution
+
+
+def check_file(path, reference, suffix='.py',
+               type_check=os.path.isfile, type_name='regular file',
+               access_check=os.R_OK, access_name='readable'):
+    if suffix is not None:
```

Comment:
As you are setting the default parameter for `suffix` different from `None`, I don't see a reason for this verification. Unless the possibility of calling the function with `suffix=None` exists.

You could choose one of the explicit options, set `suffix=None` and keep this verification or leave `suffix` as it is now and remove this verification.

Reply:
I just noticed you call it on `avocado/plugins/resolvers.py` with `suffix=None` and now that makes more sense to me :)